internal/api/handler_test.go

```diff
@@ -504,6 +504,17 @@ func Test_HandleExchangeToken(t *testing.T) {
 		_ = assertRespStatus(t, http.DefaultClient, req, http.StatusBadRequest)
 	})
 
+	t.Run("return 400 if grant type is invalid", func(t *testing.T) {
+		form := url.Values{
+			"grant_type": {"abcd"},
+		}
+		req, err := http.NewRequest(http.MethodPost, proxyTokenURL, strings.NewReader(form.Encode()))
+		require.NoError(t, err)
```

Comment:
This does not test the described case as `grant_type` is empty on the server-side. I think you need to add appropriate `Content-Type` for this to work.